providers/Tidal/mod.ts

```diff
@@ -0,0 +1,330 @@
+				break;
+			}
+			offset += limit;
+			query.set('offset', String(offset));
+			this.updateCacheTime(timestamp);
+		}
+
+		return tracklist;
+	}
+
+	protected async convertRawRelease(rawRelease: Album): Promise<HarmonyRelease> {
+		this.id = rawRelease.id;
+		const rawTracklist = await this.getRawTracklist(this.id);
+		const media = this.convertRawTracklist(rawTracklist);
+		return {
+			title: rawRelease.title,
+			artists: rawRelease.artists.map(this.convertRawArtist.bind(this)),
+			gtin: rawRelease.barcodeId,
+			externalLinks: [{
+				url: new URL(rawRelease.tidalUrl),
+				types: ['paid streaming'],
+			}],
+			media,
+			releaseDate: parseHyphenatedDate(rawRelease.releaseDate),
+			copyright: rawRelease.copyright,
+			status: 'Official',
+			packaging: 'None',
+			images: this.getLargestCoverImage(rawRelease.imageCover),
+			labels: this.getLabels(rawRelease),
+			info: this.generateReleaseInfo(),
+		};
+	}
+
+	private convertRawTracklist(tracklist: AlbumItem[]): HarmonyMedium[] {
+		const result: HarmonyMedium[] = [];
+		let medium: HarmonyMedium = {
+			number: 1,
+			format: 'Digital Media',
+			tracklist: [],
+		};
+
+		// Get info about video tracks to show a warning to the user.
+		const videoTrackInfo: string[] = [];
+
+		// split flat tracklist into media
+		tracklist.forEach((item) => {
+			// store the previous medium and create a new one
+			if (item.volumeNumber !== medium.number) {
+				if (medium.number) {
+					result.push(medium);
+				}
+
+				medium = {
+					number: item.volumeNumber,
+					format: 'Digital Media',
+					tracklist: [],
+				};
+			}
+
+			if (item.artifactType === 'video') {
+				videoTrackInfo.push(`${item.trackNumber}: ${item.title}`);
+			}
+
+			medium.tracklist.push(this.convertRawTrack(item));
+		});
+
+		if (videoTrackInfo.length) {
+			this.addMessage(
+				`This release contains ${videoTrackInfo.length} ${plural(videoTrackInfo.length, 'video track')}:\n- ${
+					videoTrackInfo.join('\n- ')
+				}`,
+				'warning',
+			);
+		}
+
+		// store the final medium
+		result.push(medium);
+
+		return result;
+	}
+
+	private convertRawTrack(track: AlbumItem): HarmonyTrack {
+		const result: HarmonyTrack = {
+			number: track.trackNumber,
+			title: track.title,
+			length: track.duration * 1000,
+			isrc: track.isrc,
+			artists: track.artists.map(this.convertRawArtist.bind(this)),
+		};
+
+		return result;
+	}
+
+	private convertRawArtist(artist: SimpleArtist): ArtistCreditName {
+		return {
+			name: artist.name,
+			creditedName: artist.name,
+			externalIds: this.provider.makeExternalIds({ type: 'artist', id: artist.id.toString() }),
+		};
+	}
+
+	private getLargestCoverImage(images: Image[]): Artwork[] {
+		let largestImage: Image | undefined;
+		let maxSize = 0;
+		images.forEach((i) => {
+			if (i.width > maxSize) {
+				largestImage = i;
+				maxSize = i.width;
+			}
+		});
+		if (!largestImage) return [];
+		return [{
+			url: new URL(largestImage.url),
```

Comment:
There is also `thumbUrl` which is displayed on the frontend by default to save bandwidth, usually with a size of about 200 or 250 pixels.

Reply:
Added support for this in c828fdc2b9240c7d1ffca73b9c77998f7c51fda2